Formula/grafana.rb

```diff
@@ -0,0 +1,123 @@
+
+    cd grafana_path do
+      # Might do it differently for head vs. release
+      system %q(sed -i.bak 's/"sass-lint": "^1.6.0",/"sass-lint": "1.7.0",/;' package.json)
+      system "go", "run", "build.go", "setup"
+      system "go", "run", "build.go", "build"
+      system "npm", "install", *Language::Node.local_npm_install_args
+      system "npm", "install", "grunt-cli", *Language::Node.local_npm_install_args
+      system "node_modules/grunt-cli/bin/grunt", "build"
+    end
+
+    cp(grafana_path/"conf/sample.ini", grafana_path/"conf/grafana.ini")
+
+    bin.install grafana_path/"bin/grafana-cli"
+    bin.install grafana_path/"bin/grafana-server"
+    (etc/"grafana").mkpath
+    (var/"log/grafana").mkpath
+    (var/"lib/grafana").mkpath
+    (var/"lib/grafana/plugins").mkpath
+    etc.install grafana_path/"conf/grafana.ini" => "grafana/grafana.ini"
+    pkgshare.install Dir[grafana_path/"conf", grafana_path/"public_gen", grafana_path/"vendor"]
+    mv pkgshare/"public_gen", pkgshare/"public"
+  end
+
+  plist_options :manual => "grafana-server --config=#{HOMEBREW_PREFIX}/etc/grafana/grafana.ini --homepath #{HOMEBREW_PREFIX}/share/grafana cfg:default.paths.logs=#{HOMEBREW_PREFIX}/var/log/grafana cfg:default.paths.data=#{HOMEBREW_PREFIX}/var/lib/grafana cfg:default.paths.plugins=#{HOMEBREW_PREFIX}/var/lib/grafana/plugins"
+
+  def plist; <<-EOS.undent
+    <?xml version="1.0" encoding="UTF-8"?>
+    <!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
+    <plist version="1.0">
+      <dict>
+        <key>KeepAlive</key>
+        <dict>
+          <key>SuccessfulExit</key>
+          <false/>
+        </dict>
+        <key>Label</key>
+        <string>#{plist_name}</string>
+        <key>ProgramArguments</key>
+        <array>
+          <string>#{opt_bin}/grafana-server</string>
+          <string>--config</string>
+          <string>#{HOMEBREW_PREFIX}/etc/grafana/grafana.ini</string>
+          <string>--homepath</string>
+          <string>#{HOMEBREW_PREFIX}/share/grafana</string>
+          <string>cfg:default.paths.logs=#{var}/log/grafana</string>
+          <string>cfg:default.paths.data=#{var}/lib/grafana</string>
+          <string>cfg:default.paths.plugins=#{var}/lib/grafana/plugins</string>
+        </array>
+        <key>RunAtLoad</key>
+        <true/>
+        <key>WorkingDirectory</key>
+        <string>#{var}/share/grafana</string>
+        <key>StandardErrorPath</key>
+        <string>#{var}/log/grafana/grafana-stderr.log</string>
+        <key>StandardOutPath</key>
+        <string>#{var}/log/grafana/grafana-stdout.log</string>
+        <key>SoftResourceLimits</key>
+        <dict>
+          <key>NumberOfFiles</key>
+          <integer>10240</integer>
+        </dict>
+      </dict>
+    </plist>
+   EOS
+  end
+
+  test do
+    system bin/"grafana-server", "-v"
+    RUBY_VERSION =~ /^(\d+\.\d+)/
+    v = $1.to_f
+    Dir.mktmpdir("grafana", HOMEBREW_TEMP) do |tdir|
```

Comment:
I added a temp directory because the test was blowing up without it, but I shifted the steps around in the coming commit & was able to remove that. Thanks for pointing it out.